README.md

```diff
@@ -0,0 +1,49 @@
+# Introduction
+
+The **hal_gigadevice** is a set of standard firmware library from GigaDevice.
+
+# How to submit code
+
+ - When submiting an updated firmware library version, it is important make
+ sure that the last firmware version will be merged. The library version can
+ be checked at [GigaDevice MCU Site](http://www.gd32mcu.com/en/download/).
+ - This repository is managed mainly on Linux. This means that changes should
+ be submited using Linux LF format. Usually running a similar command to
+ `find * type f -print0 | xargs -0 dos2unix -k` should be enough.
+ - Make sure directory structure is respected.
+ - Directory names should be converted to lowercase.
+ - All funtions must be prefixed with `gd32_`.
+ - All defines and enum values must be prefixed with `GD32_`.
```

Comment:
does this mean all functions/definitions/enums? If so, that needs to be scripted somehow.

Reply:
After I sent I noted that all is to wide. Try restrict with public before but yes, it is a bit effort.

Reply:
This rule definitely needs to come with a script to automate this process, otherwise additions or updates will cost a lot of work, not to mention that it will be an error-prone process. The vendor should also be made aware so that they can improve their HAL naming scheme (I'm pretty sure we're not the first users facing this issue).

Reply:
Not necessarily. It is possible add a grace period and adopt initially fix only current conflicts with rules defined here until a whole solution be implemented. It can be made gradually too without a driver or even mandatory for drivers that are implemented in tree. Once we define how this will work we'll stop conflicts at review phase and everybody know what need to do. I don't have intention to add a dependency like a script that may block development.

A script is the best solution and as soon it exists the whole solution can be implemented but shouldn't be a block on my view.

Reply:
So I think the initial approach should be to fix conflicting names, and once a script is in place extend to every public item. Having this bit in the readme without HALs actually doing it will create confusion.